### PR TITLE
gh-108364: Simplify quoting values and identifiers in sqlite3's iterdump()

### PR DESCRIPTION
**Pull request by felixxm**
Cleanup for #108340.

* Issue: gh-108364

### CODE REVIEW

File: Lib/sqlite3/dump.py [hunk 1]
```diff
@@ -7,6 +7,14 @@
 # future enhancements, you should normally quote any identifier that
 # is an English language word, even if you do not have to."
 
+def quote_name(name):
+    return '"{0}"'.format(name.replace('"', '""'))
+
+
+def quote_value(value):
+    return "'{0}'".format(value.replace("'", "''"))
```
Comment (felixxm) [inline Lib/sqlite3/dump.py]: I wasn't sure if there is any policy about public/private hooks. Should I start names with `_`?

Comment (erlend-aasland) [reply Lib/sqlite3/dump.py]: Yeah, that's best, IMO. We don't want APIs added accidentally. Another option is to explicitly `del` them after use.

**Comment by erlend-aasland** [comment]
This is a nice cleanup. I think we should add this as a separate PR, instead of as a part of #108340.

File: Lib/sqlite3/dump.py (hunk 1 above)
Comment (felixxm) [reply Lib/sqlite3/dump.py]: Changed :+1:

**Comment by felixxm** [comment]
> This is a nice cleanup. I think we should add this as a separate PR, instead of as a part of #108340.

Agreed. #108340 could use new hooks so I would merge this PR first.

**Comment by erlend-aasland** [comment]
> Agreed. #108340 could use new hooks so I would merge this PR first.

Sounds good to me.

**Comment by erlend-aasland** [comment]
Seems `main` was recently broken by #107988; we'll have to wait for that issue to be resolved before landing.

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**